src/main/java/com/datapipe/jenkins/vault/credentials/VaultLdapCredential.java

```diff
@@ -0,0 +1,58 @@
+package com.datapipe.jenkins.vault.credentials;
+
+import com.bettercloud.vault.Vault;
+import com.bettercloud.vault.VaultConfig;
+import com.bettercloud.vault.VaultException;
+import com.cloudbees.plugins.credentials.CredentialsScope;
+import com.cloudbees.plugins.credentials.impl.BaseStandardCredentials;
+import com.datapipe.jenkins.vault.exception.VaultPluginException;
+import hudson.Extension;
+import hudson.util.Secret;
+import org.kohsuke.stapler.DataBoundConstructor;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+
+public class VaultLdapCredential extends BaseStandardCredentials implements VaultCredential {
+
+    private final @Nonnull String serviceAccountName;
+    private final @Nonnull
```

Comment:
Please rebase, and `NonNull` from FindBugs/SpotBugs
The annotation should be on its own line
```suggestion
    @NonNull
    private final Secret serviceAccountPassword;
```